kymatio/backend/tensorflow_backend.py

```diff
@@ -1,49 +1,54 @@
 import numpy as np
+import tensorflow as tf
 
 
-class TensorFlowBaseBackend():
-    def __init__(self, tf):
-        self.tf = tf
+class Modulus():
+    """This class implements a modulus transform for complex numbers.
 
-    def complex_check(self, x):
-        if not self. _is_complex(x):
-            raise TypeError('The input should be complex.')
-    
-    def real_check(self, x):
-        if not self._is_real(x):
-            raise TypeError('The input should be real.')
-    
-    def _is_complex(self, x):
-        return (x.dtype == np.complex64) or (x.dtype == np.complex128)
-    
-    
-    def _is_real(self, x):
-        return (x.dtype == np.float32) or (x.dtype == np.float64)
-    
-    
-    class Modulus():
-        """This class implements a modulus transform for complex numbers.
-    
-        Parameters
-        ----------
-        x: input complex tensor.
-    
-        Returns
-        ----------
-        output: a complex tensor equal to the modulus of x.
-    
-        Usage
-        ----------
-        modulus = Modulus()
-        x_mod = modulus(x)
-        """
-        def __call__(self, x):
-            norm = tf.abs(x)
-            return norm
-    
-    
-    def concatenate(self, arrays, dim):
-        return self.tf.stack(arrays, axis=dim)
+    Parameters
+    ----------
+    x: input complex tensor.
+
+    Returns
+    ----------
+    output: a complex tensor equal to the modulus of x.
+
+    Usage
+    ----------
+    modulus = Modulus()
+    x_mod = modulus(x)
+    """
+    def __call__(self, x):
+        norm = tf.abs(x)
+        return norm
+
+
+def complex_check(x):
+    if not _is_complex(x):
+        raise TypeError('The input should be complex.')
+
+
+def real_check(x):
+    if not _is_real(x):
+        raise TypeError('The input should be real.')
+
+
+def _is_complex(x):
+    return (x.dtype == np.complex64) or (x.dtype == np.complex128)
+
+
+def _is_real(x):
+    return (x.dtype == np.float32) or (x.dtype == np.float64)
+   
+ 
+class TensorFlowBackend():
+    def __init__(self):
+        self.name = 'tensorflow'
+        self.modulus = Modulus()
+  
+
+    def concat(self, arrays, dim):
```

Comment:
Why not `concatenate` like we have elsewhere?

Reply:
so we call this function in the other files. 

before, when we had the namedtuple convention we would be able to do this:
`backend.concatenate = lambda x: concatenate(x, -2)`

Now, from the inheritance of this backend here, we have to worry about this:

`def concatenate(self, x):
    return self.concatenate(x, -2)`

this makes the code a bit scary, I don't know whats going on, additionally it was crashing with this convention I described above.

Reply:
I don't follow

Reply:
The second one won't work since inheritance in Python clobbers the method in the superclass (I think). How about this?
```
def contatenate(self, arrays):
    return tf.stack(arrays, dim=1)
```
The way we've set things up, the second dimension (`dim = 1`) is always the coefficient dimension, so we always want to concatenate along there, independent of the dimension of the signals. So there's only one `concatenate` that is used by all the backends.

Reply:
3D is a world unto itself, I'm afraid. It may warrant a special `concatenate_3d` that does something different.